core/src/main/java/org/apache/calcite/rel/core/Aggregate.java

```diff
@@ -322,15 +324,24 @@ public static RelDataType deriveRowType(RelDataTypeFactory typeFactory,
     assert groupList.size() == groupSet.cardinality();
     final RelDataTypeFactory.FieldInfoBuilder builder = typeFactory.builder();
     final List<RelDataTypeField> fieldList = inputRowType.getFieldList();
+    final Set<String> containedNames = Sets.newHashSet();
     for (int groupKey : groupList) {
-      builder.add(fieldList.get(groupKey));
+      final RelDataTypeField field = fieldList.get(groupKey);
+      containedNames.add(field.getName());
+      builder.add(field);
     }
     if (indicator) {
       for (int groupKey : groupList) {
         final RelDataType booleanType =
             typeFactory.createTypeWithNullability(
                 typeFactory.createSqlType(SqlTypeName.BOOLEAN), false);
-        builder.add("i$" + fieldList.get(groupKey).getName(), booleanType);
+        String name = "i$" + fieldList.get(groupKey).getName();
+        int i = 0;
+        while (containedNames.contains(name)) {
+          name += "_" + i++;
+        }
```

Comment:
Do you really intend creating long column names?
Can you replace `name +=` with `name = base + "_" + i++`?